src/RulesEngine/Validators/RuleValidator.cs

```diff
@@ -37,7 +37,7 @@ public RuleValidator()
 
         private void RegisterExpressionTypeRules()
         {
-            When(c => c.Operator == null && c.RuleExpressionType == RuleExpressionType.LambdaExpression, () => {
+            When(c => c.Operator == null && (c.Rules == null ? true : c.Rules.Count() != 0) && c.RuleExpressionType == RuleExpressionType.LambdaExpression, () => {
                 RuleFor(c => c.Expression).NotEmpty().WithMessage(Constants.LAMBDA_EXPRESSION_EXPRESSION_NULL_ERRMSG);
                 RuleFor(c => c.Rules).Null().WithMessage(Constants.LAMBDA_EXPRESSION_RULES_ERRMSG);
```

Comment:
Adding `(c.Rules == null ? true : c.Rules.Count() != 0)` in when condition will skip validation if there are nested rules.

I would recommend modifying line 42 with:
``` c#
RuleFor(c => c.Rules).Empty().WithMessage(Constants.LAMBDA_EXPRESSION_RULES_ERRMSG);
```

Also please add unit tests for this change. It will help us to not break this in future 😄